### PR TITLE
implemented proper delete/re-add of plugins

### PR DESCRIPTION
**Pull request by EricWittmann**
...as well as upgrading of plugins to newer versions

### CODE REVIEW

File: manager/ui/hawtio/plugins/api-manager/ts/forms/edit-gateway.ts [hunk 1]
```diff
@@ -89,6 +89,7 @@ module Apiman {
                 $scope.updateButton.state = 'in-progress';
                 var gateway = Gateway();
                 ApimanSvcs.update({ entityType: 'gateways', secondaryType: $scope.gateway.id }, gateway, function() {
+                    $rootScope.isDirty = false;
```
Comment (msavy) [inline manager/ui/hawtio/plugins/api-manager/ts/forms/edit-gateway.ts]: Is this a stowaway change? Doesn't seem related :)

Comment (EricWittmann) [inline manager/ui/hawtio/plugins/api-manager/ts/forms/edit-gateway.ts]: Yeah I slipped a bug fix in there too.  Good catch. :)

Comment (EricWittmann) [inline manager/ui/hawtio/plugins/api-manager/ts/forms/edit-gateway.ts]: @kahboom - this was the fix to the edit-gateway page.  The bug was that I was getting the "you haven't saved your changes yet!" dialog when clicking on the 'Save' button.  The reason is that the 'Save' button on that page results in a redirect of the user to a new page.  So the solution was to set the isDirty flag to false if the save worked, prior to doing the redirect. :)

Comment (kahboom) [inline manager/ui/hawtio/plugins/api-manager/ts/forms/edit-gateway.ts]: @EricWittmann - Thanks for tagging me on this. That was a good catch, I hope we don't have that issue in other areas. I'll do some additional testing later today. :)

Comment (EricWittmann) [inline manager/ui/hawtio/plugins/api-manager/ts/forms/edit-gateway.ts]: I think that's the only place.  But only because many of the other "edit"
pages don't implement isDirty functionality at all.  :)  We can add that to
the ever growing list of improvements to make.
